strawberryfields/tdm/tdmprogram.py

```diff
@@ -557,6 +557,8 @@ def compile(self, *, device=None, compiler=None):
                                     device.target, program_param, param_range
                                 )
                             )
+            self._compile_info = (device, "TDM")
+            self._target = device.target
```

Comment:
Should we update the docstring to state that compilation also sets the target and compilation information fields of the TDM program? Otherwise, it may seem that calling `TDMProgram.compile()` is always an optional step.

Reply:
It is an optional (manual) step since the `Engine` will call the compile method internally when run (this is the same way that the X8 compilation happens). This happens both on the `LocalEngine` and `RemoteEngine`, compiling either for the simulator backend or the target device.

Reply:
Updated the docstring slightly still, since it makes sense to have it there. 🙂